submissions/kt-std/friends-app/js/script.js

```diff
@@ -0,0 +1,447 @@
+let FRIENDS_ARRAY = [],
+    INITIAL_FRIENDS_ARRAY = [];
+const USERS_AMOUNT = 24,
+    API_URL = `https://randomuser.me/api/?results=${USERS_AMOUNT}`,
+    MIN_AGE_INPUT_ID = "minAge",
+    MAX_AGE_INPUT_ID = "maxAge",
+    CARDS_CONTAINER = document.querySelector(".cards__container"),
+    SELECT_CONTAINER = document.querySelector(".select__container"),
+    FILTERS_CONTAINER = document.querySelector(".filters__container"),
+    OPTIONS_CONTAINER = document.querySelector(".select__list"),
+    TOTAL_COUNTER = document.querySelector(".amount");
+
+function getFriends() {
+    fetch(API_URL)
+        .then((response) => {
+            if (response.ok) {
+                return response.json();
+            } else {
+                appendErrorMessage(
+                    getResponseErrorMessage(
+                        response.status,
+                        response.statusText
+                    )
+                );
+            }
+        })
+        .then((responseBody) => {
+            if (responseBody !== undefined) {
```

Comment:
> The Promise returned from fetch() won’t reject on HTTP error status even if the response is an HTTP 404 or 500. Instead, it will resolve normally (with ok status set to false), and it will only reject on network failure or if anything prevented the request from completing. ([MDN](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch))

In my case, when `!response.ok`, it returns nothing, just appending an error message on the screen. If necessary, I can modify the code by adding `return response.ok` or `return false`, which are equal in that case, and then change it to `if(responseBody)`.

Reply:
Just throw error when `!response.ok`

Reply:
Works like a charm! :)